### ISSUE TITLE
net/haproxy plugin: Adding resolver to real server template ends with error

### ISSUE THREAD

**Issue by wszychta** [opnsense/plugins#2340]
Important notices
Before you add a new report, we ask you kindly to acknowledge the following:

[X] I have read the contributing guide lines at https://github.com/opnsense/plugins/blob/master/CONTRIBUTING.md

[X] I have searched the existing issues and I'm convinced that mine is new.

[X] When the request is meant for an existing plugin, I've added its name to the title.

**Describe the bug**
I wanted to use `Real Servers template` option which is getting hosts information from SRV record. To do this I have created Resolver which is pointing to my local DNS server (outside of OPNSense instance). When I wanted to check this setup I got an error:
```
[NOTICE] 109/075727 (15720) : haproxy version is 2.2.11-c58c4e4
[ALERT] 109/075727 (15720) : parsing [/usr/local/etc/haproxy.conf.staging:132]: 'resolve-prefer' expects either ipv4 or ipv6 as argument.
[ALERT] 109/075727 (15720) : Error(s) found in configuration file : /usr/local/etc/haproxy.conf.staging
[ALERT] 109/075727 (15720) : Fatal errors found in configuration.
```

I have exported text configuration and I see that for some reasons haproxy plugin is adding `resolve-prefer` option without any arguments at the end of `server-template`

I did this for the first time, so I can't tell if this problem exists on previous versions of the plugin. I believe that the problem exists since plugin version 3.0 when `* add support for resolve-prefer option (#1975)` was fixed.

You can read more about this option in [documentation](https://cbonte.github.io/haproxy-dconv/2.0/configuration.html#resolve-prefer)

**To Reproduce**
1. Create Resolver with default options and correct DNS servers
2. Create desired SRV records in your DNS server 
3. Create Real server with type template
4. Use defined resolver with default options
5. Click Save and Test syntax button
6. See error

**Expected behavior**
1. Haproxy should not add at the end of `server-template` option `resolve-prefer` or setup it in correct way
2. All checks should pass

**Additional context**
Changed parts of my configuration file:
```
+# Resolver: test_dns
+resolvers 607db889e443c2.90753878
+    nameserver 1.2.3.4:53 1.2.3.4:53
+    nameserver 4.3.2.1:53 4.3.2.1:53
+    resolve_retries 3
+    timeout resolve 1s
+    timeout retry 1s

+# Backend: test_backend
+backend test_backend
+    # health check: test_monitor
+    mode tcp
+    balance source
+    # stickiness
+    stick-table type ip size 50k expire 30m  
+    stick on src
+    # tuning options
+    timeout connect 30s
+    timeout server 30s
+    server-template test 3 test.example.com:6443 check inter 2s  resolvers 607db889e443c2.90753878 resolve-prefer 
```

**Environment**

OPNSense 21.1.4
plugin os-haproxy version 3.1
System was installed from ISO on VM with 1 core and 2GB of RAM

--

**Comment by wszychta** [comment]
Now I have detected that problem also exists when not using `Real Server template`
```
+# Backend: test_api_backend
+backend test_api_backend
+    # health check: test_api_monitor
+    mode tcp
+    balance source
+    # stickiness
+    stick-table type ip size 50k expire 30m  
+    stick on src
+    # tuning options
+    timeout connect 30s
+    timeout server 30s
+    server test1_6443 192.168.0.21:6443 check inter 2s  resolve-prefer 
+    server test2_6443 192.168.0.22:6443 check inter 2s  resolve-prefer 
+    server test3_6443 192.168.0.23:6443 check inter 2s  resolve-prefer 
+
+# Backend: test_80_backend
+backend test_80_backend
+    # health checking is DISABLED
+    mode tcp
+    balance source
+    # stickiness
+    stick-table type ip size 50k expire 30m  
+    stick on src
+    # tuning options
+    timeout connect 30s
+    timeout server 30s
+    server test1_80 192.168.0.31:80 resolve-prefer 
+    server test2_80 192.168.0.32:80 resolve-prefer 
+    server test3_80 192.168.0.33:80 resolve-prefer 
+    server test4_80 192.168.0.34:80 resolve-prefer 
```

**Comment by fraenki** [comment]
>     server test1_80 192.168.0.31:80 resolve-prefer 

Thanks for the report. This was indeed a bug, caused by a simple copy and paste error in the template.

This issue will be fixed in the upcoming release 3.4 of os-haproxy.
